RoutesApp_ios/modules/lineList/viewmodels/RouteListViewModel.swift

```diff
@@ -8,117 +8,67 @@
 import Foundation
 
 class RouteListViewModel: ViewModel {
-    var lineRouteManager: LineRouteManagerProtocol = LineRouteFirebaseManager.shared
-    var lineCategoryManager: LineCategoryManagerProtocol = LineCategoryFirebaseManager.shared
-    var lineManager: LineManagerProtocol = LineFirebaseManager.shared
+    var localDataManager: LocalDataManagerProtocol = LocalDataManager.shared
 
-    var routeListModel: [Lines] = []
-    var linesCategory: [LinesCategory] = []
-    var lineRouteList: [LineRouteInfo] = []
-    private var originalList = [RouteListDetailModel]()
-    var filteredRouteList: [RouteListDetailModel] = []
-    var filteredByCategoryRouteList: [RouteListDetailModel] = []
-    var filteredByQueryRouteList: [RouteListDetailModel] = []
-    var fecthedLineRoute = { () -> Void in}
-       var lineFetched: Bool = false {
-           didSet {
-               fecthedLineRoute()
-           }
-       }
-    var selectedFilterIndex = -1
+    var lines = [LineEntity]()
+    private var originalLines = [LineEntity]()
+    var categories = [LineCategoryEntity]()
 
-    func getLines(completion: @escaping () -> Void) {
-        lineManager.getLinesForCurrentCity { result in
+    var queryAux = ""
+    var categoryAux: LineCategoryEntity?
+
+    func getLines() {
+        localDataManager.getDataFromCoreData(type: LineEntity.self, forEntity: LineEntity.name) { result in
```

Comment:
Are you getting data by cityId? because if you change from city it will get from all the cities or how are you handling that ?

Reply:
ah i see, that's good